mintpy/ifgram_inversion.py

```diff
@@ -1095,17 +1095,20 @@ def ifgram_inversion(ifgram_file='ifgramStack.h5', inps=None):
 
         # Look at the ~/.config/dask/mintpy.yaml file for Changing the Dask configuration defaults
 
-        # This line submits NUM_WORKERS jobs to Pegasus to start a bunch of workers
+        # This line submits NUM_WORKERS jobs to the cluster to start a bunch of workers
         # In tests on Pegasus `general` queue in Jan 2019, no more than 40 workers could RUN
         # at once (other user's jobs gained higher priority in the general at that point)
+        #
+        # When running as a local cluster, we want to use all available cpus on the computing node, 
+        # which is accssible via the `multiprocessing` stndard library.
         NUM_WORKERS = inps.numWorker
+        if inps.cluster == 'local':
+            import multiprocessing as mpc
+            NUM_WORKERS = mpc.cpu_count()
```

Comment:
I am hesitant of using all CPUs on the local machine. I usually prefer to let code run while I am doing other stuff. How about we change the default `numWorker` value for the `local` cluster to 4 and keep it default as 40 for the other cluster types?

One way to do it is adding here https://github.com/insarlab/MintPy/blob/master/mintpy/utils/utils1.py#L650:

```
cluster_key = 'mintpy.networkInversion.cluster'
cluster = templateDict.get(cluster_key, templateAutoDict[cluster_key])
if cluster == 'local':
    templateAutoDict['mintpy.networkInversion.numWorker'] = 4
```

Then here we could have:

```
inps.numWorker = min(inps.numWorker, multiprocessing.cpu_count())
NUM_WORKERS = inps.numWorker
```

Reply:
Problem is, Falk wants it to use all the CPUs without having to specify how many there are.

Reply:
We could add a flag option for whether to use all available resources on the machine, which then uses `mpc.cpu_count()` and otherwise require a specified number of workers.

Reply:
It's fine to use all CPUs. We could add that translation, i.e. `mintpy.networkInversion.numWorker = all` equals to `multiprocessing.cpu_count()`. 

It's not okay to set this as default for everyone.

Reply:
I prefer the following, let's do not create a new option unless we have to.

mintpy.networkInversion.numWorker   = auto #[int > 0 / all], auto is 4 for local cluster and 40 for the other types, number of works to deploy, all to use all available resources (for local only).